### PR TITLE
Matching Diacritics

### PR DESCRIPTION
**Pull request by joshuabyler**
Ran into an issue where if a charcter had diacritics (accents), then it does not match.  Added a function that matches characters with diacritics to the matching character without it.  Also wrote a test to make sure that it is working.

### CODE REVIEW

**Comment by vlazar** [comment]
Thanks for contribution! It's super cool you've added tests for new feature.

Some notes from quick glimpse into the code:
- you've used `text.latinise()` in `FILTER_CONTAINS` and `FILTER_STARTSWITH`, but not in `SORT_BYLENGTH`
- you've only replaced diacritics in item's text, but not input value; what if user inputs text with diacritics?
- the `String.prototype` should not be modified, `latinise()` can be a local function
- there are a lot of unrelated formatting changes, which makes this PR harder to read, please keep existing formatting style

I'm not sure, if this approach would work for every language. E.g will sorting work correctly in any language? Different languages have different rules. Internationalization is hard.

@LeaVerou Do you think diacritics should be handled in Awesomplete itself by default?

We might want to look into https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl in the near feature, as browser support is not quite there yet.

**Comment by joshuabyler** [comment]
I'm really sorry about all the stupid formatting changes, my ide just changed all of that crap on me.

You make a great point about the input needing to be cleaned up too, I could add that, and I could add the function to the sort by length function.  I'll also remove latinise() from the String prototype.  Should have those changes soon.

Also, about people entering different languages, you're right about i18n being really hard.  It's tough figuring out all the different possible things that could be entered, and then making sure they match.  I was just trying to address diacritics, but if there is an option that would fix it for different languages too that would be awesome

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules
 coverage
+.idea
```
Comment (vlazar) [inline .gitignore:3]: This can be in a local git ignore, since it's not for all users.

**Comment by vlazar** [comment]
Don't put to much effort into this yet. Let's discuss first.

I think you can clean up formatting and remove latinise() from the String prototype and wait for Lea's comments on this, since she is the project owner. I can't decide by myself if this will be merged, or not.

**Comment by joshuabyler** [comment]
cool thanks for the quick response

**Comment by vlazar** [comment]
> Also, about people entering different languages, you're right about i18n being really hard. It's tough figuring out all the different possible things that could be entered, and then making sure they match.

And we probably should not do it. For example [this Intl polyfill is 30k minified](https://github.com/andyearnshaw/Intl.js)

**Comment by LeaVerou** [comment]
Sorry to be a spoilsport, but I think this should be a plugin.
(no strong opinion though, I'm open to being convinced otherwise)
Thanks for the work @heronblade!
We should have a plugin directory, as the "advanced examples" list is getting too long.

**Comment by joshuabyler** [comment]
hey no problem, i completely understand.

I'm still going to clean it up, and try and make it better.  By the way @LeaVerou awesomplete is pretty freaking cool, it has been really nice working with it, and in fact the company I work for just released it on Steinway's new Artist [page](http://steinway.com/music-and-artists)

**Comment by LeaVerou** [comment]
Thank you @heronblade, glad you like it! And definitely do work on it, this is a common request, so we'll use your code one way or another. Even if it doesn't make it to the core library, it will definitely be added as a plugin, once we have a plugin directory.

> I can't decide by myself if this will be merged, or not.

Just noticed this. For small, noncontroversial PRs, you can (this one is neither though :)

File: .gitignore (hunk 1 above)
Comment (eliasgonzalezugalde) [reply .gitignore:3]: How can I enable this feature? I need that awesomplete ignore ñ - á - é - í - ó - ú, etc

**Pull request closed without merging**